tests/component-tests/textfieldControl/js/TextfieldSliderTests.js

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2009 University of Toronto
+Copyright 2011-2016 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+*/
+
+/* global fluid, jqUnit */
+
+(function ($) {
+    "use strict";
+
+    $(document).ready(function () {
+
+        jqUnit.module("TextfieldSlider Tests");
+
+        fluid.defaults("fluid.tests.textfieldSlider", {
+            gradeNames: ["fluid.textfieldSlider"],
+            ariaOptions: {
+                "aria-labelledby": "label-nativeHTML"
+            },
+            strings: {
+                "aria-label": "Aria self-labeling"
+            }
+        });
+
+        jqUnit.test("Test Init", function () {
+            jqUnit.expect(10);
+            var options = {
+                model: {
+                    value: 15,
+                    range: {
+                        min: 10,
+                        max: 20
+                    }
+                }
+            };
+            var that = fluid.tests.textfieldSlider(".fl-textfield-slider", options);
+
+            fluid.tests.textfieldControl.assertRangeControlledTextfieldInit(that.textfield, options);
+
+            var slider = that.locate("slider");
+            jqUnit.assertEquals("The value now should be " + options.model.value, options.model.value, +slider.val());
+            jqUnit.assertEquals("The max should be " + options.model.range.max, options.model.range.max, +slider.attr("max"));
+            jqUnit.assertEquals("The min should be " + options.model.range.min, options.model.range.min, +slider.attr("min"));
+            jqUnit.assertEquals("Slider has user-supplied aria-label value", that.options.strings["aria-label"], slider.attr("aria-label"));
+            jqUnit.assertEquals("Slider has user-supplied aria-labelled value", that.options.ariaOptions["aria-labelledby"], slider.attr("aria-labelledby"));
+        });
+
+        fluid.tests.textfieldSlider.assertTextfieldEntry = function (valToTest, expected, that) {
```

Comment:
This actually sets the value through the textfield. A similar thing is done for the slider in another function.